core/components/header/header-section.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import { useParams } from 'next/navigation';
+import { ComponentProps, useCallback } from 'react';
+
+import { HeaderSection as VibesHeaderSection } from '@/vibes/soul/sections/header-section';
+import { usePathname, useRouter } from '~/i18n/routing';
+
+type Props = ComponentProps<typeof VibesHeaderSection>;
+
+const useSwitchLocale = () => {
+  const pathname = usePathname();
+  const router = useRouter();
+  const params = useParams();
+
+  return useCallback(
+    (locale: string) =>
+      router.push(
```

Comment:
Using `next-intl`'s [recommended way](https://next-intl.dev/docs/routing/navigation#userouter) to switch between locales, with the exception of pushing the URL change to history (`router.push` vs `router.replace`).